docs/aad-e2e/aad-e2e.md

```diff
@@ -0,0 +1,234 @@
+</div>
+
+- open the app -> API permissions, pls add the permission in the picture following. If the azure active directly is created by yourself and you will manage the group and user info by yourself, you could grant the permission by yourself. Or you should apply the permission through your aad's admin.
+ 
+<div  align="center">
+<img src="image/api_permissions.png" alt="paictl overview picture" style="float: center; margin-right: 10px;" />
+</div>
+
+- open the app -> quickstart, then get the following value:
+    - Application (client) ID, mark it as ```${app_id}```
+    - Directory (tenant) ID, mark it as ```${tenant_id}```
+    - In the endpoints' view, you will find the OpenID Connect metadata document and mark it as ```${well_known}```
+
+<div  align="center">
+<img src="image/quick_start.png" alt="paictl overview picture" style="float: center; margin-right: 10px;" />
+</div>
+
+##### Configure authentication part in service-configuration.yaml
+
+```yaml
+authentication:
+  OIDC: true
+  OIDC-type: AAD
+  AAD:
+   wellKnownURL: ${well_known}
+   tenantID: ${tenant_id}
+   clientID: ${app_id}
+   clientSecret: ${secret_value}
+   nonceLifetime: null
+   nonceMaxAmount: 10
+   clockSkew: null
+  
+  group-manager:
+    group-data-source: ms-graph
+  
+    # Admin group name and its user list
+    admin-group:
+      # The group named showed in OpenPAI system.
+      groupname: admingroup
+      description: "admin's group"
+      # The group alias (groupname) in Azure Active directory
+      externalName: "team_alias_a"
+    
+    # Group for default vc.
+    # For yarn default queue hack.
+    default-group:
+      # The group named showed in OpenPAI system.
+      groupname: default
+      description: "group for default vc"
+      # The group alias (groupname) in Azure Active directory
+      externalName: "team_alias_b"
+    
+    # If you cluster you have configured several yarn vc, except default vc (it has been created in the default-group), you should configure group for each vc in the following list
+    grouplist:
+    # The group named showed in OpenPAI system.
+    - groupname: forexample1
+      description: forexample1
+    # The group alias (groupname) in Azure Active directory
+      externalName: "team_alias_c"
+    # The group named showed in OpenPAI system.
+    - groupname: forexample2
+      description: forexample2
+    # The group alias (groupname) in Azure Active directory
+      externalName: "team_alias_d"    
+```
+
+##### Optinal. Migrate restserver from basic mode to azure active directory.
+
+- Please clean all user's data. Because in this mode, user's permission will be managed by azure active directory. The local data is useless.
+    - ```sudo ./paictl.py service delete -n rest-server```
+- If server vc has been configured in yarn, pls remember configure them in grouplist in authentication.
+
+##### After all the steps above, you could following our tutorial to start openpai.
+
+#### Test
```

Comment:
if test fails, what's the recommendation?

Reply:
Done. If fail several times, creating issue is recommended